pkg/catalogv2/git/download.go

```diff
@@ -1,124 +1,89 @@
 package git
 
 import (
-	"crypto/sha256"
-	"encoding/hex"
-	"encoding/pem"
 	"fmt"
-	"net/url"
-	"os"
-	"path/filepath"
-	"regexp"
-	"strings"
-
-	"github.com/rancher/rancher/pkg/settings"
-	corev1 "k8s.io/api/core/v1"
 )
 
-const (
-	stateDir  = "management-state/git-repo"
-	staticDir = "/var/lib/rancher-data/local-catalogs/v2"
-	localDir  = "../rancher-data/local-catalogs/v2" // identical to helm.InternalCatalog
-)
-
-func gitDir(namespace, name, gitURL string) string {
-	staticDir := filepath.Join(staticDir, namespace, name, hash(gitURL))
-	if s, err := os.Stat(staticDir); err == nil && s.IsDir() {
-		return staticDir
+// Ensure will check if repo is cloned, if not the method will clone and reset to the latest commit.
+// If reseting to the latest commit is not possible it will fetch and try to reset again
+func (r *repository) Ensure(branch string) error {
+	err := r.cloneOrOpen("")
+	if err != nil {
+		return err
 	}
-	localDir := filepath.Join(localDir, namespace, name, hash(gitURL))
-	if s, err := os.Stat(localDir); err == nil && s.IsDir() {
-		return localDir
+
+	// Try to reset to the given commit, if success exit
+	localBranchFullName := fmt.Sprintf("refs/heads/%s", branch)
```

Comment:
Why are we introducing the localBranch here ? The old code just passes the branch right ? and we need to replicate the old behaviour. Any new changes should come with an issue created and then separate PR.